avocado/plugins/run.py

```diff
@@ -215,7 +215,8 @@ def configure(self, parser):
         settings.add_argparser_to_option('job.run.store_logging_stream',
                                          parser=parser.output,
                                          long_arg='--store-logging-stream',
-                                         metavar='LOGGING_STREAM')
+                                         metavar='LOGGING_STREAM',
+                                         argparse_type=lambda x: set(x.split(',')))
```

Comment:
What about adding this method inside `key_type` during registration?

Reply:
> What about adding this method inside `key_type` during registration?

Wouldn't that mess up the parsing of the config files?

Reply:
Your method should be aware of that and handle that as well, checking the type. I remember seeing some options.

Reply:
> Your method should be aware of that and handle that as well, checking the type. I remember seeing some options using it that way.

Can you give one example? I don't think we have a case where the syntax on the config is parsed one way, and in the command line it's parsed another way.

Reply:
> Sure, `avocado/plugins/variants.py:map_verbosity_level()` is not exactly doing this with config/command parse but it is prepared to accept two different types.
> 

Right, it accepts two data types (either int or string), but the same *input* (be it a string or int) is good for both the config file and command line entry.  The problem with `--store-logging-stream` is quite different, because the input we *must* give on the config file, just won't work on the command line.

> My point is that, maybe we could try using the already in place property "key_type" because the current logic will support it and we will avoid needing two methods for complex situations. Adding a new argument could make things a little bit more confusing.
> 

Are you suggesting that we should, say attempt to parse the content as a Python list, and if it fails, assume a comma separated string and split it?

> Regardless of your choice this seems to me that we have to revisit this "list" type. Maybe we should even replace type="list" by something else, since we have the "nargs" + "append" option inside argparse.

I agree... there should be one obvious way to handle lists.  I did not go with that route because currently:

1. there's an expected behavior on `--store-logging-stream`
2. It's related with `--show`

It won't make sense to ask a user to do:

```
avocado --show-test,console run --store-logging-stream test --store-logging-stream console -- ...
```

IMO the next step is to discuss the list behavior on the command line, deprecate the current behavior, and eventually switch to a new behavior.

Reply:
> > Sure, `avocado/plugins/variants.py:map_verbosity_level()` is not exactly doing this with config/command parse but it is prepared to accept two different types.
> 
> Right, it accepts two data types (either int or string), but the same _input_ (be it a string or int) is good for both the config file and command line entry. The problem with `--store-logging-stream` is quite different, because the input we _must_ give on the config file, just won't work on the command line.

I see.

 
> > My point is that, maybe we could try using the already in place property "key_type" because the current logic will support it and we will avoid needing two methods for complex situations. Adding a new argument could make things a little bit more confusing.
> 
> Are you suggesting that we should, say attempt to parse the content as a Python list, and if it fails, assume a comma separated string and split it?

That could be one option, but tbh I haven't thought much about the consequences.

> 
> > Regardless of your choice this seems to me that we have to revisit this "list" type. Maybe we should even replace type="list" by something else, since we have the "nargs" + "append" option inside argparse.
> 
> I agree... there should be one obvious way to handle lists. I did not go with that route because currently:
> 
>     1. there's an expected behavior on `--store-logging-stream`
> 
>     2. It's related with `--show`
> 
> 
> It won't make sense to ask a user to do:
> 
> ```
> avocado --show-test,console run --store-logging-stream test --store-logging-stream console -- ...
> ```
> 
> IMO the next step is to discuss the list behavior on the command line, deprecate the current behavior, and eventually switch to a new behavior.

Sure, Not only on command-line but also on config files.I have created a new issue to discuss that:

https://github.com/avocado-framework/avocado/issues/4838

Thanks for the explanation.